src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/FindFinancialTransactionCategoryIT.java

```diff
@@ -0,0 +1,114 @@
+package pl.byczazagroda.trackexpensesappbackend.integration;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.User;
+import pl.byczazagroda.trackexpensesappbackend.model.UserStatus;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.UserRepository;
+import pl.byczazagroda.trackexpensesappbackend.service.UserService;
+
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class FindFinancialTransactionCategoryIT extends BaseIntegrationTestIT {
+
+    @Autowired
+    FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    private UserService userService;
+
+    private String validAccessToken;
+
+    @BeforeEach
+    public void setup() {
+        userRepository.deleteAll();
+        financialTransactionCategoryRepository.deleteAll();
+
+        User user = new User();
+        user.setUserName("test");
+        user.setPassword("password");
+        user.setEmail("test@example.com");
+        user.setUserStatus(UserStatus.VERIFIED);
+        userRepository.save(user);
+
+        validAccessToken = userService.createAccessToken(user);
+    }
```

Comment:
This piece of code can cause a race condition and basicly wrong output.
Tests in spring boot run simultaneously and when two @BeforeEach functions try to write to validAccessToken it will be race condition.
Another issue with that is u can get same access token in multiple tests (the last one created) so it means that u have access token with id of possibly not existing user.
Try to replace it to not set class field inside of @BeforeEach method
U can try using this aproach instead
```java
mockMvc.perform(MockMvcRequestBuilders.post(<String: link>)
                        .with(SecurityMockMvcRequestPostProcessors.user(<String: user_id>)))
```
or move the token creation part inside of each test method